core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceTemplateParameterConverter.java

```diff
@@ -58,14 +60,26 @@ public GcpInstanceTemplateV1Parameters convert(GcpInstanceTemplateV4Parameters s
         return new GcpInstanceTemplateV1Parameters();
     }
 
-    public AzureInstanceTemplateV4Parameters convert(AzureInstanceTemplateV1Parameters source) {
+    public AzureInstanceTemplateV4Parameters convert(AzureInstanceTemplateV1Parameters source, DetailedEnvironmentResponse environment) {
         AzureInstanceTemplateV4Parameters response = new AzureInstanceTemplateV4Parameters();
         response.setEncrypted(source.getEncrypted());
+        initEncryption(response, environment);
         response.setManagedDisk(source.getManagedDisk());
         response.setPrivateId(source.getPrivateId());
         return response;
     }
 
+    private void initEncryption(AzureInstanceTemplateV4Parameters response, DetailedEnvironmentResponse environment) {
+        // FIXME Fetch DES ID from environment; environment can be null!
```

Comment:
environment can not be null as your outer converter checks for that with the getIfNotNull

Reply:
Correct, the code has been updated to reflect this.